### ISSUE TITLE
Exception when saving sections with Super Table fields in Content Migration

### ISSUE THREAD

**Issue by mgburns** [engram-design/SuperTable#146]
### Description

I'm attempting to write a content migration that updates a section that contains a static Super Table field. When I attempt to save the section the following exception is thrown:

```txt
Exception: Calling unknown method: verbb\supertable\elements\SuperTableBlockElement::all() (/var/www/html/vendor/yiisoft/yii2/base/Component.php:300)
#0 /var/www/html/vendor/craftcms/cms/src/base/Element.php(790): yii\base\Component->__call('all', Array)
#1 /var/www/html/vendor/verbb/super-table/src/fields/SuperTableField.php(561): craft\base\Element->__call('all', Array)
#2 /var/www/html/vendor/craftcms/cms/src/services/Content.php(236): verbb\supertable\fields\SuperTableField->getSearchKeywords(Object(verbb\supertable\elements\SuperTableBlockElement), Object(craft\elements\Entry))
#3 /var/www/html/vendor/craftcms/cms/src/services/Content.php(201): craft\services\Content->_updateSearchIndexes(Object(craft\elements\Entry), Object(craft\models\FieldLayout))
#4 /var/www/html/vendor/craftcms/cms/src/services/Elements.php(459): craft\services\Content->saveContent(Object(craft\elements\Entry))
#5 /var/www/html/vendor/craftcms/cms/src/services/Sections.php(1145): craft\services\Elements->saveElement(Object(craft\elements\Entry))
#6 /var/www/html/vendor/craftcms/cms/src/services/Sections.php(542): craft\services\Sections->_onSaveSingle(Object(craft\models\Section), false, Array)
#7 /var/www/html/migrations/m180407_225100_rename_section_with_static_super_table.php(26): craft\services\Sections->saveSection(Object(craft\models\Section))
#8 /var/www/html/vendor/craftcms/cms/src/db/Migration.php(42): craft\contentmigrations\m180407_225100_rename_section_with_static_super_table->safeUp()
#9 /var/www/html/vendor/yiisoft/yii2/console/controllers/BaseMigrateController.php(725): craft\db\Migration->up()
#10 /var/www/html/vendor/yiisoft/yii2/console/controllers/BaseMigrateController.php(199): yii\console\controllers\BaseMigrateController->migrateUp('m180407_225100_...')
#11 /var/www/html/vendor/craftcms/cms/src/console/controllers/MigrateController.php(274): yii\console\controllers\BaseMigrateController->actionUp(0)
#12 [internal function]: craft\console\controllers\MigrateController->actionUp(0)
#13 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#14 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#15 /var/www/html/vendor/yiisoft/yii2/console/Controller.php(148): yii\base\Controller->runAction('', Array)
#16 /var/www/html/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('', Array)
#17 /var/www/html/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate', Array)
#18 /var/www/html/vendor/yiisoft/yii2/console/Application.php(147): yii\console\Application->runAction('migrate', Array)
#19 /var/www/html/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#20 /var/www/html/craft(22): yii\base\Application->run()
#21 {main}
```

Note that if I make it a non-static Super Table field the migration succeeds.

### Steps to reproduce

1. Create a static Super Table field
2. Create a section that contains the static Super Table field in its field layout
3. Add some content to the Super Table field in the entry associated with the section and save it
3. Run a content migration that attempts to update that section

Here's a configuration that can be imported using the Architect plugin:

```json
{
    "fieldGroups": [
        "Test"
    ],
    "sections": [
        {
            "name": "Test Single",
            "handle": "testSingle",
            "type": "single",
            "enableVersioning": true,
            "siteSettings": [
                {
                    "hasUrls": "1",
                    "uriFormat": "test-single",
                    "template": "_test-single",
                    "enabledByDefault": true
                }
            ]
        }
    ],
    "fields": [
        {
            "group": "Test",
            "name": "Static Super Table",
            "handle": "staticSuperTable",
            "instructions": "",
            "type": "verbb\\supertable\\fields\\SuperTableField",
            "minRows": "",
            "maxRows": "",
            "localizeBlocks": false,
            "staticField": "1",
            "fieldLayout": "table",
            "selectionLabel": "",
            "blockTypes": [
                {
                    "name": "Text",
                    "handle": "text",
                    "instructions": "",
                    "type": "craft\\fields\\PlainText",
                    "typesettings": {
                        "placeholder": "",
                        "code": "",
                        "multiline": "",
                        "initialRows": "4",
                        "charLimit": "",
                        "columnType": "text"
                    }
                }
            ]
        }
    ],
    "entryTypes": [
        {
            "sectionHandle": "testSingle",
            "name": "Test Single",
            "handle": "testSingle",
            "hasTitleField": true,
            "titleLabel": "Title",
            "titleFormat": "{section.title}",
            "fieldLayout": {
                "Content": [
                    "staticSuperTable"
                ]
            }
        }
    ]
}
```

And here's a content migration that will trigger the exception:

```php
<?php

namespace craft\contentmigrations;

use Craft;
use craft\db\Migration;

/**
 * m180407_225100_rename_section_with_static_super_table migration.
 */
class m180407_225100_rename_section_with_static_super_table extends Migration
{
    /**
     * @inheritdoc
     */
    public function safeUp()
    {
        $sections = Craft::$app->getSections();
        $siteId = Craft::$app->sites->getPrimarySite()->id;

        echo '    > renaming "Test Single" section ...';
        $section = $sections->getSectionByHandle('testSingle');
        $section->name = 'Test Single Renamed';
        $section->handle = 'testSingleRenamed';
        $sections->saveSection($section);
        echo ' done' . PHP_EOL;
    }
}
```

### Additional info

- Plugin version: 2.0.5
- Craft version: 3.0.1

**Comment by engram-design** [comment]
Hmm, I've tested with your exact setup, and not getting any errors on my end, the section converts just fine.

```
Exception: Calling unknown method: verbb\supertable\elements\SuperTableBlockElement::all() (/var/www/html/vendor/yiisoft/yii2/base/Component.php:300)
```

Seems like the above error is struggling to the find element query. This is in particular related to updating search indexes.

Are you triggering the migration just through the control panel?

**Comment by mgburns** [comment]
In this case I was triggering migrations from the CLI. I will attempt one from the CP and report back.

FWIW, we're attempting to introduce [Schematic](https://github.com/nerds-and-company/schematic) for schema management now that it supports Craft 3 and this error is preventing imports from succeeding there as well.

**Comment by engram-design** [comment]
Ah yes, there was an issue with Static Super Table's from the CLI, which is fixed in the latest `composer require verbb/super-table:dev-craft3`. We're also using Schematic, but have other issues to sort out (namely actually integrating with it!) cc https://github.com/nerds-and-company/schematic/issues/123

**Comment by mgburns** [comment]
Awesome, thanks! Looks like we'll still need to wait a bit on Schematic, but its good to know it's on both of your radars. I'll verify that this issue is resolved in `develop` and close this accordingly.

**Comment by lenvanessen** [comment]
Whats the news on this one? I noticed that schematic added the ability so should be able to implement right?

**Comment by engram-design** [comment]
@lenvanessen Still working on the integration.

**Comment by engram-design** [comment]
This has been added/fixed in [2.0.7](https://github.com/verbb/super-table/releases/tag/2.0.7).
